src/store/actions.js

```diff
@@ -1,70 +1,123 @@
-        id: null,
-        items: [],
-        filters: {
-          tags: [],
-          categories: [],
-        },
-        checkedFilters: {
-          tags: [],
-          categories: [],
-        },
-      });
-    }
-  },
+  // local storage
 
-  // lists
+  _setListIdFromLocalStorage({ commit, dispatch }) {
+    const currentListId = JSON.parse(localStorage.getItem('currentListId'));
 
-  _setInitialState({ commit }, lists) {
-    commit('setInitialState', lists);
+    if (currentListId) {
+      commit('setCurrentListId', currentListId);
+      dispatch('_fetchListById', currentListId);
+    }
   },
-  _addList({ commit, dispatch }, list) {
-    commit('addList', list);
-
-    dispatch('_updateLocal');
+  _updateListIdInLocalStorage({ getters }) {
+    localStorage.setItem('currentListId', JSON.stringify(getters.currentListId));
   },
-  _saveList({ commit, dispatch }, list) {
-    commit('saveList', list);
 
-    dispatch('_updateLocal');
+  // lists
+
+  _fetchListsForUser({ commit, dispatch }) {
+    return this._vm.$axios.get(`${this._vm.$apiBasePath}lists`)
+      .then(response => {
+        commit('setLists', response.data);
+        dispatch('_setListIdFromLocalStorage');
+      })
+      .catch(error => {
+        throw error;
+      });
   },
-  _removeList({ commit, dispatch, getters }, listId) {
-    if (getters.currentListId === listId) {
-      const switchListId = getters.lists.length > 1
-        ? getters.lists.find(list => list.id !== listId).id
-        : null;
+  _fetchListById({ commit, dispatch, getters }, id) {
+    dispatch('_setCurrentListId', id);
 
-      commit('switchList', switchListId);
+    if (getters.currentListObj.items.length) {
+      if (getters.currentListObj.items[0] instanceof Object) {
+        dispatch('_setCurrentItems', getters.currentListObj.items);
+      }
     }
 
-    commit('removeList', listId);
-
-    dispatch('_updateLocal');
+    return this._vm.$axios.get(`${this._vm.$apiBasePath}list/${id}`)
+      .then(response => {
+        commit('updateList', response.data);
+        dispatch('_setCurrentItems', response.data.items);
+      })
+      .catch(error => {
+        throw error;
+      });
   },
-  _setListForEditting({ commit }, id) {
-    commit('setListForEditting', id);
+  _addList({ commit, dispatch }, list) {
+    return this._vm.$axios.post(`${this._vm.$apiBasePath}list/add`, list)
+      .then(response => {
+        commit('addList', response.data);
+        dispatch('_setCurrentListId', response.data.id);
+      })
+      .catch(error => {
+        throw error;
+      });
   },
-  _filterList({ commit, dispatch, getters }, filters) {
-    commit('filterList', {
-      currentListIndex: getters.currentListIndex,
-      filters,
-    });
+  async _addTestList({ commit, dispatch }, listToSend) {
+    const listDataToSend = {
+      name: listToSend.name,
+      isPrivate: listToSend.isPrivate,
+      tags: listToSend.tags,
+      categories: listToSend.categories,
+    };
```

Comment:
maybe we can destructure properties which we want in argument of this function? and here we can use short form after